### PR TITLE
Restore the native keyboard shortcuts in cells editors

### LINKED ISSUE #9367: Restore the native keyboard shortcuts in cells editors
### Description

The issue is about restoring the native behavior of some keyboard shortcuts. There are some conflicts with OS once cell editing. The issue is part of the [Epic issue](https://github.com/handsontable/handsontable/issues/9361) and here are the expected results:

Shortcut | Expected behavior
-- | --
Home | Move the cursor to the beginning of the text 
End | Move the cursor to the end of the text 
Cmd/Ctrl + Z | Undo last action entered into cell's editor 
Cmd/Ctrl + Shift + Z | Redo last action entered into cell's editor

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR changes the behavior of the Home/End, Ctrl/Cmd + Z, and Ctrl/Cmd + Shift + Z keyboard shortcuts. Now it works as follows:

Shortcut | Expected behavior
-- | --
Home | Move the cursor to the beginning of the text 
End | Move the cursor to the end of the text 
Cmd/Ctrl + Z | Undo last action entered into cell's editor 
Cmd/Ctrl + Shift + Z | Redo last action entered into cell's editor 

Some edge-cases, limitations, or additional new behavior:
 * `Home/End` keyboard shortcuts implement their own mechanism for caret movement. As research showed, it works differently on Chrome and non-Chrome browsers. The PR implements the non-Chrome approach where hitting the Home or End key on an empty line does not move the caret at all - on Chrome, it moves up/down until the caret reaches the line that contains a text (more than one character). The differences can be shown as follow:
Chrome implementation:
Kapture 2022-04-13 at 15 32 55
FF implementation (the same for Safari):
Kapture 2022-04-13 at 15 33 51

### How has this been tested?

I tested the changes on Win7 (Chrome, FF), Ubuntu LTS (FF, Chromium), and macOS (Chrome, FF, Safari) and didn't find any issues. I covered the changes with the new E2E and unit tests.

### Types of changes

- [x] New feature or improvement (non-breaking change which adds functionality)

### Related issue(s):
1. fixes #9367

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Review by kirszenbaum — APPROVED** [review]
I'll document these changes in #9271

**Review by wszymanski — CHANGES_REQUESTED** [review]
It seems that height of the editor isn't changed after undo action. It's updated just after pressing some extra key.

https://user-images.githubusercontent.com/141330/163991004-55b0b9e0-c016-41d5-ae46-45f47eacad90.mp4

edit: It works similarly for redo action.

File: handsontable/src/editors/textEditor/__tests__/caretPositioner.unit.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    textarea.remove();
+  });
+
+  describe('updateCaretPosition()', () => {
+    describe('should keep the caret position untouched when "actionName" is unknown', () => {
+      it('for non-multiline text', () => {
+        textarea.value = 'text text';
+
+        setCaretPosition(textarea, 3);
+        updateCaretPosition('endd', textarea);
```
Comment (wszymanski) [inline handsontable/src/editors/textEditor/__tests__/caretPositioner.unit.js:21]:
```suggestion
        updateCaretPosition('unknown', textarea);
```

File: handsontable/src/editors/textEditor/__tests__/caretPositioner.unit.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      });
+    });
+
+    describe('should keep the caret position untouched when "actionName" is unknown', () => {
+      it('for multiline text', () => {
+        textarea.value = `
+text`;
+
+        setCaretPosition(textarea, 1);
+        updateCaretPosition('endd', textarea);
```
Comment (wszymanski) [inline handsontable/src/editors/textEditor/__tests__/caretPositioner.unit.js:37]:
```suggestion
        updateCaretPosition('unknown', textarea);
```

**Review by wszymanski — APPROVED** [review]

**Comment by budnix** [comment]
> It seems that height of the editor isn't changed after undo action. It's updated just after pressing some extra key.

I fixed that within https://github.com/handsontable/handsontable/pull/9393/commits/c07a39e2396f00fc05efe247b5c9449139ccfa9c.

**Review by wszymanski — APPROVED** [review]

**Pull request merged**